harness/determined/common/experimental/determined.py

```diff
@@ -72,10 +72,36 @@ def _from_session(cls, session: api.Session) -> "Determined":
     def create_user(
         self, username: str, admin: bool, password: Optional[str] = None, remote: bool = False
     ) -> user.User:
+        """Creates a user.
+
+        The user's credentials may be managed by a remote service, in which case the `remote`
+        argument should be set to `true`, and then SSO should be configured for the user.
+        Otherwise, a password must be set that meets complexity requirements.
+
+        The complexity requirements are:
+            - Must be at least 8 characters long.
+            - Must contain at least one upper-case letter.
+            - Must contain at least one lower-case letter.
+            - Must contain at least one number.
+
+        Arg:
+            username: username of the user.
+            admin: indicates whether the user is an admin.
+            password: password of the user.
+            remote: indicates whether the user is managed by a remote service.
+
+        Returns:
+            A :class:`~determined.experimental.client.User` of the created user.
+
+        Raises:
+            ValueError: an error describing why the password does not meet complexity requirements.
+        """
         create_user = bindings.v1User(username=username, admin=admin, active=True, remote=remote)
         hashedPassword = None
-        if password is not None:
-            hashedPassword = api.salt_and_hash(password)
+        if not remote:
+            api.check_password_complexity(password)
+            if password is not None:  # always true provided check_password didn't raise
```

Comment:
Weirdly, I like this better without the comment. I see this as going one of two ways:
1. `check_password_complexity` guarantees that `password is not None`
2. ... _doesn't_ guarantee `password is not None`

If (1) and that's an assumption `create_user` wants to make, then this should be an assertion.

If (2), then we should make sure both cases (is None and isn't None) are covered with the right behavior and the comment is unhelpful.

I'll leave it to you to decide between the (1) and (2). I just prefer that it be either instead of this kind-of-in-between.

Reply:
I lean toward (1) personally, but the linter feels strongly about (2). Under the assumption that the linter represents the org's consensus opinion, I'll remove the comment.